chaoslib/litmus/azure-disk-loss/lib/azure-disk-loss.go

```diff
@@ -0,0 +1,287 @@
+package lib
+
+import (
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/compute/mgmt/compute"
+	experimentTypes "github.com/litmuschaos/litmus-go/pkg/azure/disk-loss/types"
+	clients "github.com/litmuschaos/litmus-go/pkg/clients"
+	diskStatus "github.com/litmuschaos/litmus-go/pkg/cloud/azure/disk"
+	instanceStatus "github.com/litmuschaos/litmus-go/pkg/cloud/azure/instance"
+	"github.com/litmuschaos/litmus-go/pkg/events"
+	"github.com/litmuschaos/litmus-go/pkg/log"
+	"github.com/litmuschaos/litmus-go/pkg/probe"
+	"github.com/litmuschaos/litmus-go/pkg/types"
+	"github.com/litmuschaos/litmus-go/pkg/utils/common"
+	"github.com/litmuschaos/litmus-go/pkg/utils/retry"
+	"github.com/pkg/errors"
+)
+
+var (
+	err           error
+	inject, abort chan os.Signal
+)
+
+//PrepareChaos contains the prepration and injection steps for the experiment
+func PrepareChaos(experimentsDetails *experimentTypes.ExperimentDetails, clients clients.ClientSets, resultDetails *types.ResultDetails, eventsDetails *types.EventDetails, chaosDetails *types.ChaosDetails) error {
+
+	// inject channel is used to transmit signal notifications.
+	inject = make(chan os.Signal, 1)
+	// Catch and relay certain signal(s) to inject channel.
+	signal.Notify(inject, os.Interrupt, syscall.SIGTERM)
+
+	// abort channel is used to transmit signal notifications.
+	abort = make(chan os.Signal, 1)
+	// Catch and relay certain signal(s) to abort channel.
+	signal.Notify(abort, os.Interrupt, syscall.SIGTERM)
+
+	//Waiting for the ramp time before chaos injection
+	if experimentsDetails.RampTime != 0 {
+		log.Infof("[Ramp]: Waiting for the %vs ramp time before injecting chaos", experimentsDetails.RampTime)
+		common.WaitForDuration(experimentsDetails.RampTime)
+	}
+
+	//get the disk name  or list of disk names
+	diskNameList := strings.Split(experimentsDetails.VirtualDiskNames, ",")
+	if len(diskNameList) == 0 {
+		return errors.Errorf("no volume names found to detach")
+	}
+	instanceNamesWithDiskNames, err := diskStatus.GetInstanceNameForDisks(diskNameList, experimentsDetails.SubscriptionID, experimentsDetails.ResourceGroup)
+
+	if err != nil {
+		return errors.Errorf("error fetching attached instances for disks")
```

Comment:
```suggestion
		return errors.Errorf("error fetching attached instances for disks,err: %v",err)
```
Can we also print the error